runtime/v1/impl/src/main/java/com/walmartlabs/concord/runner/Main.java

```diff
@@ -99,7 +99,7 @@ public void run(RunnerConfiguration runnerCfg, Path baseDir) throws Exception {
         long t1 = System.currentTimeMillis();
 
         Path idPath = baseDir.resolve(Constants.Files.INSTANCE_ID_FILE_NAME);
-        while (!Files.exists(idPath)) {
+        while (isEmpty(idPath)) {
```

Comment:
Maybe we should wait until the file is readable *and* it can be parsed as UUID. I.e. loop until the ID parses :-)